doc/tutorials/specification-language.md

```diff
@@ -0,0 +1,97 @@
+# Configuration validation is now applies to all keys below the '/tests' path
+kdb mount test.spec spec/tests dump
+
+# An example configuration which will by default gets mounted to the 'user' namespace
+kdb mount test.conf /tests dump type
+```
+
+Elektra takes all configuration settings which the user or admin provides and parses it into a structured internal format.
+This structure is called a `KeySet` and is format agnostic (meaning that validation can be applied to any file format
+such as `ini`, `yml`, etc.). After successful parsing into the KeySet, Elektra then passes this KeySet to various
+validation plugins. You can get a picture of all plugins [here](/src/plugins). Every plugin performs validation
+of the KeySet depending on the metadata provided (more detail about this will come in the next section) and emits warnings
+or errors. If any error happened, the changes of the configuration settings are not applied.
+
+E.g. you can see
+that in the upper example we also mounted the [type](/src/plugins/type/README.md) plugin for validation checks. This for example
+can check for wrong types:
+
+```shell script
+kdb meta-set spec/tests/bool 'type' boolean
+kdb set /tests/bool Batman
+# Sorry, module type issued the error C03200:
+# Validation Semantic: The value 'Batman' of key 'user/tests/bool' could not be converted into a boolean
+```
+
+Please note as of Elektra v0.9 you cannot add an unlimited amount of plugins. You can see if this issue
+still remains in our issue tracker([#2133](https://github.com/ElektraInitiative/libelektra/issues/2133)).
+
+## Specification writing
+
+There are multiple ways how you can set constraints for configuration settings. All possibilities though set `metadata` to certain
+configuration settings (this is also done in the spec namespace). A metadata is just additional data describing the configuration setting.
+One way to set metadata is the command line tool [kdb meta-set](../help/kdb-meta-set.md) which we have used in the upper example.
+For the [validation plugin](../../src/plugins/validation/README.md) which supports regular expression checks there exists even its own
+command line tool [kdb vset](../help/kdb-vset.md).
+
+If you are writing validation for dozens of configuration settings, the command line might not be the most convenient way. You can also
+write specifications in any file format of your desire. Examples can be seen [here](../../examples/spec). In Elektra, all file formats
+can have metadata but some needed extra tinkering like any `yaml` parser since metadata are not natively supported. We recommend the `ini`
+file format as it natively supports metadata. So the upper example
+could be rewritten in an `ini` file in the following way:
+
+```ini
+[]
+mountpoint = test-specification.ni
+infos/plugins = type
+
+[bool]
+type = boolean
+```
+
+To integrate a configuration into the key database, we mount it by using the property `mountpoint` in the configuration specification.
```

Comment:
What is a property?

Reply:
Would "configuration key" be more appropriate for you?

Reply:
It should be either explained in the tutorial or in doc/help/elektra-glossary.md